mono/profiler/coverage.c

```diff
@@ -0,0 +1,922 @@
+			return FALSE;
+		*rval = end + 1;
+		return TRUE;
+	} else {
+		//FIXME how should we handle passing a value to an arg that doesn't expect it?
+		return !strcmp (arg, opt_name);
+	}
+}
+
+static void
+parse_arg (const char *arg)
+{
+	const char *val;
+
+	if (match_option (arg, "help", NULL)) {
+		usage ();
+	// } else if (match_option (arg, "zip", NULL)) {
+	// 	coverage_config.use_zip = TRUE;
+	} else if (match_option (arg, "output", &val)) {
+		coverage_config.output_filename = g_strdup (val);
+	} else if (match_option (arg, "covfilter", &val)) {
+		g_error ("not supported");
+	} else if (match_option (arg, "covfilter-file", &val)) {
+		if (coverage_config.cov_filter_files == NULL)
+			coverage_config.cov_filter_files = g_ptr_array_new ();
+		g_ptr_array_add (coverage_config.cov_filter_files, g_strdup (val));
+	} else {
+		mono_profiler_printf_err ("Could not parse argument: %s", arg);
+	}
+}
+
+static void
+parse_args (const char *desc)
+{
+	const char *p;
+	gboolean in_quotes = FALSE;
+	char quote_char = '\0';
+	char *buffer = malloc (strlen (desc));
+	int buffer_pos = 0;
+
+	for (p = desc; *p; p++){
+		switch (*p){
+		case ',':
+			if (!in_quotes) {
+				if (buffer_pos != 0){
+					buffer [buffer_pos] = 0;
+					parse_arg (buffer);
+					buffer_pos = 0;
+				}
+			} else {
+				buffer [buffer_pos++] = *p;
+			}
+			break;
+
+		case '\\':
+			if (p [1]) {
+				buffer [buffer_pos++] = p[1];
+				p++;
+			}
+			break;
+		case '\'':
+		case '"':
+			if (in_quotes) {
+				if (quote_char == *p)
+					in_quotes = FALSE;
+				else
+					buffer [buffer_pos++] = *p;
+			} else {
+				in_quotes = TRUE;
+				quote_char = *p;
+			}
+			break;
+		default:
+			buffer [buffer_pos++] = *p;
+			break;
+		}
+	}
+
+	if (buffer_pos != 0) {
+		buffer [buffer_pos] = 0;
+		parse_arg (buffer);
+	}
+
+	g_free (buffer);
+}
+
+static void
+usage (void)
+{
+	mono_profiler_printf ("Mono coverage profiler");
+	mono_profiler_printf ("Usage: mono --profile=coverage[:OPTION1[,OPTION2...]] program.exe\n");
+	mono_profiler_printf ("Options:");
+	mono_profiler_printf ("\thelp                 show this usage info");
+
+	mono_profiler_printf ("\tcovfilter=ASSEMBLY   add ASSEMBLY to the code coverage filters");
+	mono_profiler_printf ("\t                     prefix a + to include the assembly or a - to exclude it");
+	mono_profiler_printf ("\t                     e.g. covfilter=-mscorlib");
+	mono_profiler_printf ("\tcovfilter-file=FILE  use FILE to generate the list of assemblies to be filtered");
+	mono_profiler_printf ("\toutput=FILENAME      write the data to file FILENAME (the file is always overwritten)");
+	mono_profiler_printf ("\toutput=+FILENAME     write the data to file FILENAME.pid (the file is always overwritten)");
+	mono_profiler_printf ("\toutput=|PROGRAM      write the data to the stdin of PROGRAM");
+	mono_profiler_printf ("\t                     %%t is substituted with date and time, %%p with the pid");
```

Comment:
not sure which line this relates to

Reply:
something I removed in the code, and forgot to remove in the `usage`